### ISSUE TITLE
LoRA A and B param.requires_grad showing as false during training

### ISSUE THREAD

**Issue by RonanKMcGovern** [huggingface/transformers#26098]
### System Info

I'm running the latest version of transformers in a Colab notebook.

I'm simply printing out the param.requires_grad in the model's named parameters to see if it is set True or false, and I'm finding that all have requires_grad set to false. If there is some other list showing the LoRA params with required gradients, how can I access that?

```
base_model.model.model.embed_tokens.weight: False
base_model.model.model.layers.0.self_attn.q_proj.weight: False
base_model.model.model.layers.0.self_attn.q_proj.lora_A.default.weight: False
base_model.model.model.layers.0.self_attn.q_proj.lora_B.default.weight: False
base_model.model.model.layers.0.self_attn.k_proj.weight: False
base_model.model.model.layers.0.self_attn.k_proj.lora_A.default.weight: False
base_model.model.model.layers.0.self_attn.k_proj.lora_B.default.weight: False`
...etc.
```

### Who can help?

@SunMarc @younesbelkada 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```
        model = PeftModel.from_pretrained(base_model, path)
        model.gradient_checkpointing_enable()
        model = prepare_model_for_kbit_training(model)
        model.to(input_ids.device)

        # Print trainable parameters here
        for name, param in model.named_parameters():
            print(f"{name}: {param.requires_grad}")
```

### Expected behavior

I would have expected that all of the LoRA A and B matrixes would show that they require gradiants...

**Comment by SunMarc** [comment]
Hi @RonanKMcGovern, this happens because you used `prepare_model_for_kbit_training` after loading the adapters. `prepare_model_for_kbit_training` should be used before adding the adapters to the quantized model. Another solution would be to fix `prepare_model_for_kbit_training` so that we only freeze the base model layers.

**Comment by RonanKMcGovern** [comment]
ah thanks @SunMarc , I re-ran and now am just seeing the base layer frozen and True for the LoRA A and B
